AOCL: add v5.0

amd-toolchain-support opened the pull request:
What is new in AOCL 5.0:

**AOCL-Data Analytics (aocl-da)**
- New library for optimized data analysis and machine learning.

**AOCL-BLAS(amdblis)**
- Specific tuning for Turin and AVX512 improvements for various APIs.
- Enhanced performance and additional support in aocl_gemm add-on.

**AOCL-Compression(aocl-compression)**
- Improved single-threaded performance for ZLIB and BZIP2.
- Multi-threaded support and performance boosts for LZ4, ZSTD, ZLIB, and Snappy.
- New options for faster compression and decompression.

**AOCL-Cryptography(aocl-crypto)**
- OpenSSL Provider support and fixes for several algorithms.
- New features like Chacha20-Poly1305 Cipher and RSA enhancements.
- Performance improvements for various cryptographic functions.

**AOCL-FFTW(amdfftw)**
- Default support for Wisdom feature with a specific option.

**AOCL-LAPACK(amdlibflame)**
- Enhancements to SVD, Factorization, and Solver APIs.
- New runtime options and improved documentation.

**AOCL-LibM(amdlibm)**
- New vector APIs and variants for mathematical functions.
- Beta version of CMAKE build system for Linux.

**AOCL-LibMem(aocl-libmem)**
- New functions and improved AVX2 performance.

**AOCL-ScaLAPACK(amdscalapack)**
- Build without test framework and enabled tracing/logging for all APIs.

**AOCL-Sparse(aocl-sparse)**
- New APIs and support for various matrix operations.
- Performance improvements and multi-threading support.

**AOCL-Utils(aocl-utils)**
- New APIs for ISA details and CPU feature checks.
- Thread pinning and support for Linux and Windows.

amd-toolchain-support committed:
AOCL: add v5.0

amd-toolchain-support committed:
Merge branch 'develop' into aocl/aocl-5.0

amd-toolchain-support committed:
amd-aocl: Fix spack audit reported issues

amd-toolchain-support commented:
Requesting review from @tldahlgren who has reviewed our previous AOCL PRs

bernhardkaindl commented on var/spack/repos/builtin/packages/aocl-crypto/package.py:
```diff
@@ -35,24 +32,34 @@ class AoclCrypto(CMakePackage):
+
+    # Removed dependency on lsb_release
+    patch(
+        "lsb_release.patch",
+        sha256="b61d6d2518276c56d37e8c64d18488081af70f29a62f315ecbd23664e0e440b9",
```
@amd-toolchain-support, this patch does not apply to `aocl-crypto-4.2`:
```py
==> Installing aocl-crypto-4.2-ewtpd5ip36d36iio6jzluxri5rtuml5q [12/12]
==> No binary for aocl-crypto-4.2-ewtpd5ip36d36iio6jzluxri5rtuml5q found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/2b/2bdbedd8ab1b28632cadff237f4abd776e809940ad3633ad90fc52ce225911fe.tar.gz
1 out of 1 hunk FAILED -- saving rejects to file cmake/CompilerLinux.cmake.rej
Reversed (or previously applied) patch detected!  Assume -R? [n]
```
```suggestion
        "lsb_release.patch",
        sha256="b61d6d2518276c56d37e8c64d18488081af70f29a62f315ecbd23664e0e440b9",
        when="@5.0",     
```

amd-toolchain-support replied:
The issue has been addressed in the most recent commit.

bernhardkaindl replied:
Confirmed visually and works for the build!

bernhardkaindl commented on var/spack/repos/builtin/packages/aocl-libmem/package.py:
```diff
@@ -55,7 +54,7 @@ class AoclLibmem(CMakePackage):
     )
 
-    depends_on("cmake@3.15:", type="build")
+    depends_on("cmake@3.22:", type="build")
```
`cmake@3.30.5` cannot parse the `tools/validator/CMakeLists.txt:76` and 3 other lines:
```py
==> Installing aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli [45/87]
==> No binary for aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli found: installing from source
==> Fetching https://github.com/amd/aocl-libmem/archive/5.0.tar.gz
==> No patches needed for aocl-libmem
==> aocl-libmem: Executing phase: 'cmake'
==> [2024-10-15-16:44:05.989204] '/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/cmake-3.30.5-bz2fjg27di5kg7b5bpielyt3xnw7axrc/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli/lib;/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli/lib64' '-DCMAKE_PREFIX_PATH:STRING=/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/cmake-3.30.5-bz2fjg27di5kg7b5bpielyt3xnw7axrc;/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gmake-4.4.1-pgyr2oulfowxw7teicooufd4axmlkrpv;/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gcc-runtime-13.1.0-25qo3ld7klulv5juopmjnk73qte743xz' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DENABLE_LOGGING:BOOL=OFF' '-DENABLE_TUNABLES:BOOL=OFF' '-DBUILD_SHARED_LIBS:BOOL=ON' '-DALMEM_ARCH:STRING=auto' '/home/bkaindl/spack.6/.stage/spack-stage-aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli/spack-src'
-- The C compiler identification is GNU 13.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/bkaindl/spack.6/lib/spack/env/gcc/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
ERMS Feature Enabled on Build machine.
Native-Compiling for AVX2 Arch...
CMake Error at tools/validator/CMakeLists.txt:76 (math):
  math cannot parse the expression: "16.5 * 1024 * 1024": syntax error,
  unexpected exp_NUMBER, expecting end of file (5).


CMake Error at tools/validator/CMakeLists.txt:87 (math):
  math cannot parse the expression: "ERROR": syntax error, unexpected end of
  file (5).


CMake Error at tools/validator/CMakeLists.txt:91 (math):
  math cannot parse the expression: "ERROR * 3 / 4 - 256": syntax error,
  unexpected exp_TIMES (7).


CMake Error at tools/validator/CMakeLists.txt:92 (math):
  math cannot parse the expression: "ERROR * 3 / 4 + 256": syntax error,
  unexpected exp_TIMES (7).


-- Configuring incomplete, errors occurred!
==> Error: ProcessError: Command exited with status 1:
    '/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/cmake-3.30.5-bz2fjg27di5kg7b5bpielyt3xnw7axrc/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli/lib;/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli/lib64' '-DCMAKE_PREFIX_PATH:STRING=/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/cmake-3.30.5-bz2fjg27di5kg7b5bpielyt3xnw7axrc;/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gmake-4.4.1-pgyr2oulfowxw7teicooufd4axmlkrpv;/home/bkaindl/spack.6/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gcc-runtime-13.1.0-25qo3ld7klulv5juopmjnk73qte743xz' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DENABLE_LOGGING:BOOL=OFF' '-DENABLE_TUNABLES:BOOL=OFF' '-DBUILD_SHARED_LIBS:BOOL=ON' '-DALMEM_ARCH:STRING=auto' '/home/bkaindl/spack.6/.stage/spack-stage-aocl-libmem-5.0-liskeffbxw4oe2meduvq7inmj2dixvli/spack-src'
```
Maybe the maximum supported `cmake` version could be needed of the math is not supported by some versions?

amd-toolchain-support replied:
Looks like this is because CMake does handle floating point arithmetic only integer, and our testing never hit a CPU + `lspcu` combination that returned a fractional number of MB as a cache size.   A fix is in progress.

amd-toolchain-support replied:
The issue has been addressed with a cmake patch in the most recent commit.

bernhardkaindl replied:
Confirmed fixed using a rebuild on the same odd VM (48 cores)!

bernhardkaindl commented on var/spack/repos/builtin/packages/aocl-da/package.py:
```diff
@@ -0,0 +1,123 @@
+            depends_on(f"aocl-sparse@={vers} +shared", when="+shared")
+            depends_on(f"aocl-sparse@={vers} ~shared", when="~shared")
+
+    depends_on("amdblis threads=openmp", when="+openmp")
+    depends_on("amdlibflame threads=openmp", when="+openmp")
+    depends_on("amdblis threads=none", when="~openmp")
+    depends_on("amdlibflame threads=none", when="~openmp")
+    depends_on("aocl-sparse +openmp", when="+openmp")
+    depends_on("aocl-sparse ~openmp", when="~openmp")
```
```py
[+] /usr (external glibc-2.35-spkgpfgkrsiezsjbgpao5m6ddj2hchhj)
[+] /usr (external tar-1.34-xptu4ddqeb2i2dmtaoptlfj4w5pp42z3)
[+] /usr (external curl-7.81.0-mmqzissv47rrkvhiss7tva6d43tg734q)
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gcc-runtime-13.1.0-25qo3ld7klulv5juopmjnk73qte743xz
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/ncurses-6.5-2yuat3jia57sfgfoc7fdsl7arajm7tfe
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/xz-5.4.6-mturqa5up3ztik5n5d2vrpqwkv5smix7
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libffi-3.4.6-xqu3yeo5z7azukj7nkqfvjzmixhp44dy
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/bzip2-1.0.8-qqzuq3npoxz37mviu3f2r6xnydsy6wca
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gmake-4.4.1-krhvoy5tc5wo3p2bpim66lm35gbmykuj
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openblas-0.3.25-ncdjrufxi6opfzpdn44xdy5t5h2zlobe
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libmd-1.0.4-mvecdr3afwkcfeolgt7qsbt5uf4pwecz
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/util-linux-uuid-2.40.2-bqq7v2o43bwegp76qc2oce2xmbcwrduq
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/readline-8.2-hifil7lrxuysqa45pht5cn26tbb37lv7
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/zlib-ng-2.2.1-nljc3vayp2squ2cwoh3slf65pixhfgup
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libbsd-0.12.2-cp37gsuedh4xwswccbnh7cyjtdr6fy4j
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gdbm-1.24-ptbirawqapmetkjjxrzi3lcrjf7nb2e6
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/sqlite-3.46.0-cxlbbl5ip5qttiy5d5bkzggnqusrnbca
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/libxml2-2.10.3-yoaam5srasicodmvl6n5oqdhh2bb5ex6
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.2-h2ywktvqoxjietl6cf4pay63uties47m
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/openssl-3.3.1-ffc4thhgxyf3iuekmmqunlfdr3nh7dge
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/expat-2.6.3-wch5dt2lba7wfls3liws7iwqdfkral5y
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gettext-0.22.5-euruzl5pzhipmpgebc7fsnrnfcnkk3xj
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/python-3.13.0-jwxgsnsxwgi7rm3hiy5mviqzzvvhw6m5
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy
[+] /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc
==> Installing aocl-da-5.0-wc2rhmrxyc7o7khqv73gj4y4ghqjyuoi [28/28]
==> No binary for aocl-da-5.0-wc2rhmrxyc7o7khqv73gj4y4ghqjyuoi found: installing from source
==> Using cached archive: /var/spack/cache/_source-cache/archive/34/3458adc7be39c78a08232c887f32838633149df0a69ccea024327c3edc5a5c1d.tar.gz
==> No patches needed for aocl-da
==> aocl-da: Executing phase: 'cmake'
==> [2024-10-15-16:31:43.559779] Find (not recursive): /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib ['libaoclutils.a']
==> [2024-10-15-16:31:43.560080] Find complete: /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib ['libaoclutils.a']
==> [2024-10-15-16:31:43.560572] Find (not recursive): /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib ['libaoclutils.a']
==> [2024-10-15-16:31:43.560692] Find complete: /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib ['libaoclutils.a']
==> [2024-10-15-16:31:43.561089] Find (not recursive): /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib ['libaoclutils.a']
==> [2024-10-15-16:31:43.561202] Find complete: /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib ['libaoclutils.a']
==> [2024-10-15-16:31:43.561877] Find (not recursive): /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy/lib ['libblis.so']
==> [2024-10-15-16:31:43.562003] Find complete: /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy/lib ['libblis.so']
==> [2024-10-15-16:31:43.562583] Find (not recursive): /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu/lib ['libflame.a']
==> [2024-10-15-16:31:43.562698] Find complete: /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu/lib ['libflame.a']
==> [2024-10-15-16:31:43.563266] Find (not recursive): /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc/lib ['libaoclsparse.a']
==> [2024-10-15-16:31:43.563372] Find complete: /opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc/lib ['libaoclsparse.a']
==> [2024-10-15-16:31:43.566046] '/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.2-h2ywktvqoxjietl6cf4pay63uties47m/bin/cmake' '-G' 'Unix Makefiles' -DCMAKE_PREFIX_PATH:STRING=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openblas-0.3.25-ncdjrufxi6opfzpdn44xdy5t5h2zlobe;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew;/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.2-h2ywktvqoxjietl6cf4pay63uties47m;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy;/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gmake-4.4.1-krhvoy5tc5wo3p2bpim66lm35gbmykuj;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gcc-runtime-13.1.0-25qo3ld7klulv5juopmjnk73qte743xz' -DUTILS_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib/libaoclutils.a' '-DUTILS_CPUID_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib/libaoclutils.a' '-DUTILS_CORE_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib/libaoclutils.a' '-DBLAS_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy/lib/libblis.so' '-DBLAS_INCLUDE_DIR=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy/include/blis' '-DLAPACK_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu/lib/libflame.a' '-DLAPACK_INCLUDE_DIR=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu/include' '-DSPARSE_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc/lib/libaoclsparse.a' '-DSPARSE_INCLUDE_DIR=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc/include' '-DBUILD_EXAMPLES:BOOL=OFF' '-DBUILD_GTEST:BOOL=ON' '-DBUILD_ILP64:BOOL=OFF' '-DBUILD_SMP:BOOL=OFF' '-DBUILD_SHARED_LIBS:BOOL=OFF' '-DBUILD_PYTHON:BOOL=OFF' '/.stage/spack-stage-aocl-da-5.0-wc2rhmrxyc7o7khqv73gj4y4ghqjyuoi/spack-src'
-- The C compiler identification is GNU 13.1.0
-- The CXX compiler identification is GNU 13.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /lib/spack/env/gcc/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /lib/spack/env/gcc/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- The Fortran compiler identification is GNU 13.1.0
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /lib/spack/env/gcc/gfortran - skipped
L-BFGS-B: Fortran Compiler ID     : GNU
L-BFGS-B: Fortran Compiler flags  : -cpp -Wno-unused-parameter
RALFit: Fortran Compiler ID       : GNU
RALFit: Fortran Compiler flags    :
-- Found Python: /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/python-3.13.0-jwxgsnsxwgi7rm3hiy5mviqzzvvhw6m5/bin/python3.13 (found version "3.13.0") found components: Interpreter
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
CMake Error at tests/unit_tests/CMakeLists.txt:110 (target_compile_definitions):
  Cannot specify compile definitions for target "pca" which is not built by
  this project.


-- Building AOCL-DA 5.0.0 (Release)
CXX Compiler ID         : GNU (gfortran) platform id: Linux
COMPILER_FLAGS_COMMON   : ;-Wall;-Wextra;$<$<COMPILE_LANGUAGE:CXX>:-Wno-sign-conversion>;;-mtune=native;-mavx2;-mfma;;;-DNDEBUG;-fopenmp-simd;

OPTIONS
Architecture flag         ARCH_FLAG............:
Building with ASan        ASAN.................: OFF
Building docs             BUILD_DOC............: OFF
Building for ILP64        BUILD_ILP64..........: OFF
Building Python bindings  BUILD_PYTHON.........: OFF
Building with OpenMP      BUILD_SMP............: OFF
Shared library build      BUILD_SHARED_LIBS....: OFF
Building with Coverage    COVERAGE.............: OFF
Debug printing            DA_LOGGING...........: OFF
Building with MemSan      MEMSAN...............: OFF
Building with Valgrind    VALGRIND.............: OFF
Vectorization reports     VECTORIZATION_REPORTS: OFF

TARGETS
Build the EXAMPLE programs  BUILD_EXAMPLES.....: OFF
Build GTEST unit-tests      BUILD_GTEST........: ON

DEPENDENT LIBRARIES
BLAS             BLAS=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy/lib/libblis.so
LAPACK           LAPACK=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu/lib/libflame.a
AOCL-Sparse      SPARSE=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc/lib/libaoclsparse.a
AOCL-Utils       UTILS=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib/libaoclutils.a
AOCL-Utils/CPUID UTILS_CPUID=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib/libaoclutils.a
-- Configuring incomplete, errors occurred!
==> Error: ProcessError: Command exited with status 1:
    '/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.2-h2ywktvqoxjietl6cf4pay63uties47m/bin/cmake' '-G' 'Unix Makefiles' -DCMAKE_PREFIX_PATH:STRING=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/openblas-0.3.25-ncdjrufxi6opfzpdn44xdy5t5h2zlobe;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew;/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/cmake-3.30.2-h2ywktvqoxjietl6cf4pay63uties47m;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy;/opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/gmake-4.4.1-krhvoy5tc5wo3p2bpim66lm35gbmykuj;/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/gcc-runtime-13.1.0-25qo3ld7klulv5juopmjnk73qte743xz' -DUTILS_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib/libaoclutils.a' '-DUTILS_CPUID_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib/libaoclutils.a' '-DUTILS_CORE_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-utils-5.0-q5ezzqfjzty5trn5vursbqovk4gp4aew/lib/libaoclutils.a' '-DBLAS_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy/lib/libblis.so' '-DBLAS_INCLUDE_DIR=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdblis-5.0-vaomrlrx3xjusdoyhvxgjkkeitdnghcy/include/blis' '-DLAPACK_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu/lib/libflame.a' '-DLAPACK_INCLUDE_DIR=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/amdlibflame-5.0-vpvp3eo3u75p4oumnuflagwwenilxozu/include' '-DSPARSE_LIB=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc/lib/libaoclsparse.a' '-DSPARSE_INCLUDE_DIR=/opt/spack/linux-ubuntu22.04-skylake/gcc-13.1.0/aocl-sparse-5.0-b4zl7dlab7mknixh7ffdjbvybdnmefyc/include' '-DBUILD_EXAMPLES:BOOL=OFF' '-DBUILD_GTEST:BOOL=ON' '-DBUILD_ILP64:BOOL=OFF' '-DBUILD_SMP:BOOL=OFF' '-DBUILD_SHARED_LIBS:BOOL=OFF' '-DBUILD_PYTHON:BOOL=OFF' '/.stage/spack-stage-aocl-da-5.0-wc2rhmrxyc7o7khqv73gj4y4ghqjyuoi/spack-src'

1 error found in build log:
     36    RALFit: Fortran Compiler ID       : GNU
     37    RALFit: Fortran Compiler flags    :
     38    -- Found Python: /opt/spack/linux-ubuntu22.04-skylake/gcc-12.3.0/python-3.13.0-jwxgsnsxwgi7rm3hiy5mviqzzvvhw6m5/bin/python3.13 (found version "3.13.0") found components: Interpreter
     39    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
     40    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
     41    -- Found Threads: TRUE
  >> 42    CMake Error at tests/unit_tests/CMakeLists.txt:110 (target_compile_definitions):
     43      Cannot specify compile definitions for target "pca" which is not built by
     44      this project.
```
Could it be that a dependency is missing here?
Note: the container/host that was trying this build does not have `pkg-config` pre-installed on the host.

amd-toolchain-support replied:
This has been root-caused, there is a typo in our CMake with similarly named targets, which only triggers in the `+gtest~examples` case.  A fix for this is in process too, thank you for catching this.

amd-toolchain-support replied:
The issue has been addressed with the patch in the most recent commit.

bernhardkaindl replied:
Confirmed, saw the fix and all generated build variants succeeded now!

tldahlgren commented on var/spack/repos/builtin/packages/aocl-da/package.py:
```diff
@@ -0,0 +1,123 @@
+    """
+
+    _name = "aocl-da"
+    homepage = "https://www.amd.com/en/developer/aocl/analytics.html"
```
I cannot confirm the homepage.  Get a `404` error.

amd-toolchain-support replied:
Thank you for bringing this issue to our attention. The homepage URL has been updated correctly.

bernhardkaindl replied:
Confirming, the new homepage URL works!

tldahlgren reviewed (commented):
FWIW. The homepage of the new package is "Not found" but I could confirm the new version sha256s *and* all of the sha256s for packages whose `url`s changed.

amd-toolchain-support committed:
changes in aocl-da, aocl-libmem, and aocl-crypto

aocl-crypto: Added when condition for lsb patch
aocl-da: Corrected the URL for the data analytics home page
       - Patch to fix cmake configuration to enable examples off but gtest
aocl-libmem: Patch to build the validator only for AuthenticAMD targets

bernhardkaindl reviewed (approved):
Build results on Ubuntu 22.04.4 LTS:
The following specs were checked (extracted from `gh pr diff`):
- `amd-aocl@5.0`
- `amdblis@4.2`
- `amdfftw@4.2`
- `amdlibflame@4.2`
- `amdlibm@4.2`
- `amdscalapack@4.2`
- `aocl-compression~bzip2~lz4~lz4hc~lzma~shared~snappy~zlib~zstd+enable_fast_math@4.2`
- `aocl-crypto+ipp@4.2`
- `aocl-da~openmp~python~shared@5.0`
- `aocl-da~examples~openmp~python~shared+gtest@5.0`
- `aocl-da~examples~openmp~python~shared+ilp64@5.0`
- `aocl-da~examples~python~shared@5.0`
- `aocl-da~examples~openmp~python@5.0`
- `aocl-da~examples~openmp~shared@5.0`
- `aocl-libmem@4.2`
- `aocl-sparse~shared@4.2`
- `aocl-utils@4.2`

These specs were installed: (`~disabled` variants are hidden using `sed`)
```py
bin/spack find --variants amd-aocl@5.0 amdblis@4.2 amdfftw@4.2 amdlibflame@4.2 amdlibm@4.2 amdscalapack@4.2 aocl-compression~bzip2~lz4~lz4hc~lzma~shared~snappy~zlib~zstd+enable_fast_math@4.2 aocl-crypto+ipp@4.2 aocl-da~openmp~python~shared@5.0 aocl-da~examples~openmp~python~shared+gtest@5.0 aocl-da~examples~openmp~python~shared+ilp64@5.0 aocl-da~examples~python~shared@5.0 aocl-da~examples~openmp~python@5.0 aocl-da~examples~openmp~shared@5.0 aocl-libmem@4.2 aocl-sparse~shared@4.2 aocl-utils@4.2 | sed 's/~[a-z]*//g;s/ [a-z0-9_]*=[a-zA-Z0-9]*//g'
-- linux-ubuntu22.04-skylake / gcc@13.1.0 -----------------------
amd-aocl@5.0 build_system=bundle
amdblis@4.2_gemm+blas+cblas+suphandling build_system=makefile libs=shared,static threads=none
amdblis@4.2_gemm+blas+cblas+suphandling build_system=makefile libs=shared,static threads=openmp
amdfftw@4.2+amd-dynamic-dispatcher-fast-planner-mpi-vader-limit+mpi+openmp+shared build_system=autotools precision=double,float
amdlibflame@4.2+lapack2flame+shared+static patches=9c62ab5 threads=openmp vectorization=auto
amdlibm@4.2 build_system=scons patches=5660ac3
amdscalapack@4.2+shared
aocl-compression@4.2+enable_fast_math decompress_fast=OFF
aocl-crypto@4.2+ipp
aocl-da@5.0+ilp64
aocl-da@5.0+gtest patches=65be59e
aocl-da@5.0+examples
aocl-libmem@4.2+shared vectorization=auto
aocl-sparse@4.2+openmp_tests
aocl-utils@4.2+shared
```
Generated by https://github.com/spack/build-quality-tools/blob/main/build_pr_changes.py
```py
~/build-quality-tools/build_pr_changes.py -ma
```

bernhardkaindl reviewed (approved):
Congratulations:

All generated build variants are successfully built now,
and I was able to resolve all conversations.

As I think this was well-reviewed, I think the next phase in the progress is to merge it for integration into spack develop! 👍

alalazo commented on var/spack/repos/builtin/packages/amd-aocl/package.py:
```diff
@@ -38,21 +39,32 @@ class AmdAocl(BundlePackage):
             depends_on(f"amdblis@={vers}")
             depends_on(f"amdfftw@={vers}")
             depends_on(f"amdlibflame@={vers}")
+            depends_on("amdlibflame ^[virtuals=blas] amdblis")
```
Where these changes validated? It turns out we have a bug in an audit that let these pass, but we never allowed `^` in depends_on directives, and these `depends_on` are treated internally as:
```python
depends_on("amdlibflame")
```
due to `deps=False` in:

https://github.com/spack/spack/blob/fc4316cafaccc591f34d6dbbeabd0cad0427b340/lib/spack/spack/directives.py#L304
